src/dql.erl

```diff
@@ -1,8 +1,69 @@
 -module(dql).
--export([prepare/1, parse/1, unparse/1, glob_match/2]).
+-export([prepare/1, parse/1, unparse/1, glob_match/2, flatten/1, unparse_metric/1]).
+
+
+-type bm() :: {binary(), [binary()]}.
+
+-type gbm() :: {binary(), [binary() | '*']}.
+
+-type time() :: {time, pos_integer(), ms | s | m | h | d | w} | pos_integer().
+
+-type resolution() :: time().
+
+-type relative_time() :: time() |
+                         now |
+                         {'after' | before, pos_integer(), time()} |
+                         {ago, time()}.
+
+
+-type range() :: {between, relative_time(), relative_time()} |
+                 {last, time()}.
+
+-type comb_fun() :: avg | min | max | sum.
+-type aggr_fun1() :: derivate.
+-type aggr_fun2() :: avg | sum | min | max.
+
+-type get_stmt() ::
+        {get, bm()} |
+        {sget, gbm()}.
+
+-type aggr_stmt() ::
+        {aggr, aggr_fun1(), statement()} |
+        {aggr, aggr_fun2(), statement(), time()}.
+
+-type cmb_stmt() ::
+        {combine, comb_fun(), [statement()]}.
+
+-type statement() ::
+        get_stmt() |
+        aggr_stmt() |
+        cmb_stmt().
+
+-type flat_aggr_fun() ::
+        {aggr, aggr_fun2(), time()} |
+        {aggr, aggr_fun1()}.
+
+-type flat_terminal() ::
+        {combine, comb_fun(), [flat_stmt()]}.
+
+-type flat_stmt() ::
+        flat_terminal() |
+        {calc, [flat_aggr_fun()], flat_terminal() | get_stmt()}.
+
+-type parser_error() ::
+        {error, binary()}.
+
+-type alias() :: term().
+
+-spec parse(string() | binary()) ->
+                   parser_error() |
+                   {ok, {select, [statement()], range(), resolution()}} |
+                   {ok, {select, [statement()], [alias()], range(), resolution()}}.
```

Comment:
Line 61 is too long:                    {ok, {select, [statement()], [alias()], range(), resolution()}}..